dfdatetime/golang_time.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+"""Golang time.Time time implementation."""
+
+import decimal
+
+from dfdatetime import definitions
+from dfdatetime import factory
+from dfdatetime import interface
+
+
+class GolangTimeEpoch(interface.DateTimeEpoch):
+  """Golang time.Time epoch."""
+
+  def __init__(self):
+    """Initializes a Golang time.Time epoch."""
+    super(GolangTimeEpoch, self).__init__(1, 1, 1)
+
+
+class GolangTime(interface.DateTimeValues):
+  """Golang time.Time timestamp.
+
+  A Golang time.Time timestamp contans the number of nanoseconds since
+  January 1, 1 UTC. Depending on the version of the timestamp, the time
+  zone is stored in minutes or seconds relative to UTC.
+
+  A serialized version 1 Golang time.Time timestamp is a 15 byte value
+  that consists of:
+
+  * byte 0 - version as an 8-bit integer.
+  * bytes 1-8 - number of seconds since January 1, 1 as a little-endian signed
+      integer.
+  * bytes 9-12 - faction of second, number of nanoseconds as a little-endian
+      signed integer.
+  * bytes 13-14 - time zone offset in minutes as a 16-bit little endian integer,
+      where -1 represents UTC.
+
+  A serialized version 1 Golang time.Time timestamp is a 16 byte value
+  that consists of:
+
+  * byte 0 - version as an 8-bit integer.
+  * bytes 1-8 - number of seconds since January 1, 1 as a little-endian signed
+      integer.
+  * bytes 9-12 - faction of second, number of nanoseconds as a little-endian
+      signed integer.
+  * bytes 13-14 - time zone offset in minutes as a 16-bit little endian integer,
+      where -1 represents UTC.
+  * byte 15 - time zone offset in seconds as an 8-bit integer.
+
+  Attributes:
+    is_local_time (bool): True if the date and time value is in local time
+  """
+
+  # The delta between January 1, 1970 (unix epoch) and January 1, 1
+  # (Golang epoch)
+  _GOLANG_TO_POSIX_BASE = (
+      ((1969 * 365) + (1969 // 4) - (1969 // 100) + (1969 // 400)) *
+      definitions.SECONDS_PER_DAY)
+
+  _EPOCH = GolangTimeEpoch()
+
+  def __init__(self,
+      seconds=None, nanoseconds=None, time_zone_offset=None,
+      time_zone_seconds=None):
+    """Initializes a Golang time.Time timestamp.
+
+    Args:
+      seconds (Optional[int]): seconds since epoch.
+      nanoseconds (Optional[int]): nanoseconds component.
+      time_zone_offset (Optional[int]): the time zone in minutes, -1 is a
+          special value for UTC (no Location set).
+      time_zone_seconds (Optional[int]): the time zone in seconds. Only valid
```

Comment:
which time zone has an offset in seconds? The rest of of the code also does not support this, are these additional changes that need to be made?

Reply:
As discussed, changed to make v2 raise an error as unsupported.